sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -55,7 +57,7 @@
 
         return analyzeBatchSentimentWithResponse(
             Collections.singletonList(new TextDocumentInput("0", text, language)), null, context)
-            .map(response -> new SimpleResponse<>(response, response.getValue().iterator().next()));
+            .map(response -> processSingleResponseErrorResult(response));
```

Comment:
nit: can use method reference instead 
```suggestion
            .map(Transforms::processSingleResponseErrorResult);
```